tests/bgp/test_bgp_dual_asn.py

```diff
@@ -0,0 +1,509 @@
+import pytest
+import time
+import logging
+import ipaddress
+import random
+import re
+
+from tests.common import constants
+from tests.common import config_reload
+from tests.common.utilities import wait_tcp_connection
+from tests.common.utilities import wait_until
+from tests.common.helpers.assertions import pytest_assert
+from tests.bgp.bgp_helpers import update_routes
+from tests.generic_config_updater.test_bgp_speaker import get_bgp_speaker_runningconfig
+from tests.generic_config_updater.gu_utils import apply_patch, expect_op_success
+from tests.generic_config_updater.gu_utils import generate_tmpfile, delete_tmpfile
+from tests.generic_config_updater.gu_utils import create_checkpoint, delete_checkpoint, rollback_or_reload
+
+
+pytestmark = [
+    pytest.mark.topology('t0'),
+]
+
+logger = logging.getLogger(__name__)
+
+BGPSLB = "BGPSLBPassive"
+BGPSLB_2 = "BGPSLBPassive2"
+BGPSLB_V6 = "BGPSLBPassiveV6"
+BGPSLB_V6_2 = "BGPSLBPassiveV62"
+BGP_SRC_ADDR_RE = "neighbor {} update-source {}"
+BGP_IP_RANGE_RE = "bgp listen range {} peer-group {}"
+NEIGHBOR_ASN_LIST = [64523, 65511]
+NEIGHBOR_PORT_LIST = [9000, 9001]
+PREFIX = "3.3.3.0/24"
+PREFIX_2 = "5.5.5.0/24"
+PREFIX_V6 = "2001:3:3:3::0/64"
+PREFIX_V6_2 = "2001:5:5:5::0/64"
+
+
+def lo_intfs(duthost, tbinfo):
+    mg_facts = duthost.get_extended_minigraph_facts(tbinfo)
+
+    lo_int = lo_int_v6 = []
+    for lo_interface in mg_facts["minigraph_lo_interfaces"]:
+        if ipaddress.ip_address(lo_interface["addr"]).version == 4:
+            lo_int = lo_interface
+        elif ipaddress.ip_address(lo_interface["addr"]).version == 6:
+            lo_int_v6 = lo_interface
+        if lo_int and lo_int_v6:
+            return lo_int, lo_int_v6
+    pytest_assert(True, "Required ipv4 and ipv6 to start the test")
+
+
+@pytest.fixture(autouse=True)
+def setup_env(duthosts, rand_one_dut_hostname):
+    """
+    Setup/teardown fixture for bgp speaker config
+    Args:
+        duthosts: list of DUTs.
+        rand_selected_dut: The fixture returns a randomly selected DuT.
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+    original_bgp_speaker_config = get_bgp_speaker_runningconfig(duthost)
+    create_checkpoint(duthost)
+
+    yield
+
+    try:
+        logger.info("Rolled back to original checkpoint")
+        rollback_or_reload(duthost)
+        # this sleep is necessary since apply config to running config need sometime.
```

Comment:
What does it mean 'apply config to running config need sometime'?
If it means the rollback operation, then the config should be there if the operation succeeds.

Reply:
Oh, my bad description. I mean after rollback_or_reload() we need wait for a short time then call get_bgp_speaker_runningconfig() to get applied running config, otherwise the rollback config sometime still not applied and we may get a wrong snap short.  Found this issue during my test.